app/routes.py

```diff
@@ -1,2 +1,278 @@
+    #if sorts value == asc then ....
+    task_response = []
+    if sort_titles:
+        if sort_titles == "asc":
+            task_by_asc = Task.query.order_by(Task.title.asc())
+            for task in task_by_asc:
+                task_response.append(task.to_python_dict())   
+        
+        if sort_titles == 'desc':
+            task_by_desc = Task.query.order_by(Task.title.desc())
+            for task in task_by_desc:
+                task_response.append(task.to_python_dict())
+        return jsonify(task_response), 200
+    
+    else:
+        tasks = Task.query.all()
+        task_response = []
+        for task in tasks:
+            #using to_json helper function 
+            task_response.append(task.to_python_dict())
+        return jsonify(task_response), 200
+
+
+
+@task_bp.route("/<task_id>", methods=["GET","PUT","DELETE"],strict_slashes=False)
+def task(task_id):
+    """
+        Input: Request to read, edit, or delete a task by a given task_id
+        Output: Returns an error if the task does not exist, returns dictionary of Task instance 
+                along with OK 200 code
+    """
+    
+    task = Task.query.get(task_id)
+    # if task is not found
+    if not task:
+        return '',404
+    
+    #checking to see if task has goal_id
+    if request.method == "GET":
+        if task.goal_id:
+            return {"task": task.to_python_dict_goal_id()}
+        else:
+            return {"task":task.to_python_dict()},200
+    
+    if request.method == 'PUT':
+        request_body = request.get_json()
+        #assigning updated values to a task instance
+        task.title = request_body['title']
+        task.description = request_body['description']
+        task.completed_at = request_body['completed_at']
+        db.session.commit()
+        return jsonify(task=task.to_python_dict()), 200
+
+    if request.method == 'DELETE':
+        db.session.delete(task)
+        db.session.commit()
+        return jsonify(details=f'Task {task.task_id} "{task.title}" successfully deleted'),200
+
+
+
+@task_bp.route("<task_id>/mark_complete",methods=["PATCH"], strict_slashes=False)
+def complete(task_id):
+    """
+        Input:  Request to mark a task complete with mark_complete in the path
+        Output: Returns an error if task does not exist, marks task completed with a datetime
+                datatype and returns updated task dictionary  
+    """
+    
+    task = Task.query.get(task_id)
+    if not task:
+        return '', 404 
+    #in order to make variable truthy we need to have datatype as value
+    task.completed_at = datetime.utcnow()
+    db.session.commit()
+    # calling helper function
+    call_slack(task)
+    
+    return make_response({"task": task.to_python_dict()}, 200)#helper function to 
+
+ 
+def call_slack(task):
```

Comment:
Love that you made a separate method for this! I think it makes the code more readable